Applications/Spire/Source/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,122 @@
+#include "Spire/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.hpp"
+#include "Nexus/Definitions/Market.hpp"
+#include "Spire/Spire/QtPromise.hpp"
+
+using Filter = Spire::FilteredOrderImbalanceIndicatorModel::Filter;
+using namespace Nexus;
+using namespace Spire;
+
+FilteredOrderImbalanceIndicatorModel::FilteredOrderImbalanceIndicatorModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source_model,
+    std::vector<Filter> filters)
+    : m_source_model(std::move(source_model)),
+      m_filters(std::move(filters)) {}
+
+OrderImbalanceIndicatorModel::SubscriptionResult
+    FilteredOrderImbalanceIndicatorModel::subscribe(
+    const boost::posix_time::ptime& start,
+    const boost::posix_time::ptime& end,
+    const OrderImbalanceSignal::slot_type& slot) {
+  auto signal = std::make_shared<Subscription>(start, end);
+  auto callback = [=] (const auto& imbalance) {
+    if(is_imbalance_accepted(imbalance)) {
+      signal->m_imbalance_signal(imbalance);
+    }
+  };
+  auto [connection, promise] = m_source_model->subscribe(start, end,
+    std::move(callback));
+  m_connections.push_back(connection);
+  return {signal->m_imbalance_signal.connect(slot),
+    promise.then([=] (auto& imbalances) {
+      return filter_imbalances(imbalances);
+    })
+  };
+}
+
+Filter Spire::make_security_list_filter(
+    const std::set<std::string>& symbol_list) {
+  return [=] (const Nexus::OrderImbalance& imbalance) {
+    return symbol_list.find(imbalance.m_security.GetSymbol()) !=
+      symbol_list.end();
+  };
+}
+
+Filter Spire::make_security_filter(const std::string& filter_string) {
+  return [=] (const Nexus::OrderImbalance& imbalance) {
+    return imbalance.m_security.GetSymbol().find(filter_string) == 0;
+  };
+}
+
+Filter Spire::make_market_list_filter(
+    const std::set<std::string>& market_list) {
+  return [=, market_database = GetDefaultMarketDatabase()] (
```

Comment:
Do not use defaults except for testing.

Reply:
What's the source for the official market list?

Reply:
Pass it in as a parameter.